api/types.proto

```diff
@@ -713,3 +713,33 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+// SecretType provides information about what kind of secret this is
+enum SecretType {
+	CONTAINER = 0 [(gogoproto.enumvalue_customname) = "ContainerSecret"];
+	NODE = 1 [(gogoproto.enumvalue_customname) = "NodeSecret"];
+}
+
+// SecretReference is the linkage between a service and a secret that it uses.
+message SecretReference {
+	// Name is the name of the secret that this reference.
+	string name = 1;
```

Comment:
The `SecretDataID` is the version ID, so it refers to a `Secret`, and the version by `SecretDataID`.  (also, the rest of the code is here:  https://github.com/diogomonica/swarmkit/compare/master...secrets-streaming, although it is a giant blob that we are trying to split out into nicer chunks.  https://github.com/diogomonica/swarmkit/commit/f774f55b9e6141e7de19e024cbbc0d2b26cdc920 specifically does some construction of the `SecretReference` object.)